internal/impl/pure/processor_processors.go

```diff
@@ -0,0 +1,118 @@
+package pure
+
+import (
+	"context"
+
+	"github.com/benthosdev/benthos/v4/internal/bundle"
+	"github.com/benthosdev/benthos/v4/internal/component/interop"
+	"github.com/benthosdev/benthos/v4/internal/component/processor"
+	"github.com/benthosdev/benthos/v4/internal/log"
+	"github.com/benthosdev/benthos/v4/internal/message"
```

Comment:
Thanks for working on this! I wonder if it would be possible to avoid using the internal APIs and implement it using `service.ExecuteProcessors` just like the `cached` processor: https://github.com/benthosdev/benthos/blob/74a82c0e425e154fd88eeabcc4723ea1a77ce205/internal/impl/pure/processor_cached.go

Reply:
@mihaitodor probably best not to, in cases like these where you're wrapping other processors I think core stuff like `try` and `catch` should be internal, it'll reduce allocs and have a small performance improvement over the public APIs. It won't be perceptible in most processors but it'll make a difference when the inner processors are mappings.

Reply:
Oh, good to know 👍